src/components/oscal-utils/OSCALLinkUtils.js

```diff
@@ -32,3 +32,16 @@ export default function resolveLinkHref(
     )
   );
 }
+
+export function getURLMediaType(url) {
+  let fileExtension = "";
+  try {
+    fileExtension = new URL(url).pathname.split(".").pop();
+    if (fileExtension === "/") {
+      fileExtension = "href";
+    }
+  } catch {
+    fileExtension = "Unknown";
+  }
+  return fileExtension.length <= 4 ? fileExtension.toUpperCase() : "Unknown";
+}
```

Comment:
```suggestion
// One could imagine that this is instead an inner function of
// `guessExtensionFromHref`. For style reasons that may or may not make sense
// but I've separated it here for ease of writing and clarity.
const getFileExtension(url) {
  try {
    // TODO: Consider new URL(url).pathname.split("/").pop().split(".").pop()
    return new URL(url).pathname.split(".").pop().trim();
  } catch {
    // console.warn("Unable to detect extension for: " + url);
    return "";
  }
}

export function guessExtensionFromHref(url) {
  const fallbackFileExtension = "Unknown";
  // TODO: Use an HTTP HEAD request to attempt to get a `Content-Type` header
  // for the underlying document. The request(s) should likely properly handle
  // redirects in order to handle DOI URLs etc. This would require refactoring
  // the length check in the return as well. In fact, doing so may replace the
  // need for an extension-based check at all. But in implementing that we
  // should remember that a HEAD is more expensive than splitting strings.
  const detectedExension = getFileExtension(url);
  const extLength = detectedExtension.length;
  // This is a guess based on typical file types. If the length is less than 2
  // or more than 4, it's probably not actually a file extension but instead we've
  // grabbed some unexpected part of the filename.
  if (extLength > 4 || extLength < 2) {
    return fallbackFileExtension;
  }
  return detectedExtension;
}
```

Reply:
I'm liking the suggestions here.

After adding the `console.warn` statement, we will see this output whenever running  `OSCALLinkUtils.test.js`in the console - I think this will be okay. It seems like `pathname.split("/").pop().split(".").pop()` proves to be better than `pathname.split(".").pop().trim()` with the case of something like. The former will return "Unknown" (which is valid), while the latter returns the output "PNG/".

I will look a little more into using an HTTP HEAD request to attempt to get a `Content-Type` header.

Reply:
I think `HEAD` requests are separate ticket (that you or I should write)

Reply:
It seems like the local linter doesn't like the `console` statement, but the linter for this repo accepts it?

Reply:
> I think `HEAD` requests are separate ticket (that you or I should write)

Sounds good, I don't mind writing this at some point today.